Report on subs by url (#1722)
alphagov/email-alert-api: docs/support-tasks.md

bundle exec rake 'support:resend_failed_emails:by_id[<email_one_id>,<email_two_id>]'
```

## Get subscriber list csv report

To see a csv report for the number of subscribers for a given list:

```bash
bundle exec rake 'report:csv_subscriber_lists[<on_date>] SLUGS=<list_slug>'
```

The date should be in ISO8601 format, for example 2020-01-01.

## Get a simple count of subscribers to a list by its URL

To see a simple count of the number of subscribers for a given list:

```bash
bundle exec rake 'report:csv_subscriber_lists[<url>]'
```
This will always report on the number of active subscriptions as of now.

You can also pass it a :active_on_datetime which will count how many active subscriptions there were on a particuar date.

> (active is defined as created after that datetime and not with an "ended_one" datetime by the given date)

Reviewer: Should this be `before`? (also you have a typo)

```suggestion
(active is defined as created before that datetime and not with an "ended_on" datetime by the given date)
```

Author: yep! thanks!

I changed the implimentation of this and missed these docs

```bash
bundle exec rake 'report:csv_subscriber_lists[<url>, <active_on_date>]'
```

> The active_on_date should be in ISO8601 format, for example 2022-03-03T12:12:16+00:00.

Reviewer: Not a blocker, but it'd be nice if this could be just a date (and then the task reports on the number of subscribers by the end of that day)

Author: Yeah the end of the day bit is a good shout, I could complicate this a bit.
As is, this will default to the start of the day 🤔 perhaps end_of_day if there's no time...

I did try a more open `Time.zone.parse` that might have given more leeway on input here but freaked out at how permissive that is. Check this out:
https://gds.slack.com/archives/CADGP8PB4/p1646314848714819

Tldr...

```
Time.zone.parse("20th of Furberry")
```
gets made into a valid date.

Author: I'm gonna part this one for the moment, let's look at extended the task with this depending on what the demand looks like?

Reviewer: Yeah I saw that descent into madness 🙈 

I hadn't realised that it'd default to the start of the day but that does make sense. Maybe just call that out in the docs and it'll do for my original thought of "I don't want to type out a timestamp when I'm on support"
